service/raft/raft_group0.cc

```diff
@@ -86,6 +87,41 @@ namespace service {
 static logging::logger group0_log("raft_group0");
 static logging::logger upgrade_log("raft_group0_upgrade");
 
+// {{{ group0_rpc Maintain failure detector subscription whenever
+// group 0 configuration changes.
+
+class group0_rpc: public service::raft_rpc {
+    service::raft_group_registry& _raft_gr;
+public:
+    using raft_rpc::raft_rpc;
+    virtual void on_configuration_change(raft::server_address_set add, raft::server_address_set del) override;
+    explicit group0_rpc(raft_group_registry& raft_gr,
+            raft_state_machine& sm, netw::messaging_service& ms,
+            raft_address_map& address_map, raft::group_id gid, raft::server_id srv_id)
+        : raft_rpc(sm, ms, address_map, gid, srv_id)
+        , _raft_gr(raft_gr)
+    {}
+};
+
+void
+group0_rpc::on_configuration_change(raft::server_address_set add, raft::server_address_set del) {
```

Comment:
I don't want to give the compiler neither choice nor responsibility about deciding whether to inline this function. It's a virtual function, it can't be inline anyway - but generally  I avoid inline definitions for anything beyond trivial. Besides, today you inline it, but tomorrow it becomes too big to follow inline - or the code is moved out  to a header file, so the function has to be deinlined, which would create a diff noise. There is value in lower indent as well. 
Generally I don't think comments like that are worth making, well, unless you'd like to get to know other people's coding habits, or you're raising a point about an implicit yet fairly widespread scylla coding convention I am not aware of.

Reply:
Honestly, I didn't even think when deciding whether to inline this code or not. I forget when I started to avoid inline definitions  - but it was long, long time ago.

Reply:
I find vim  fold markers are extremely convenient for files which contain definitions for multiple classes. Fold markers are not equivalent to syntax folding, so moving the definition inline will not achieve the desired effect.

Reply:
> I don't want to give the compiler neither choice nor responsibility about deciding whether to inline this function. It's a virtual function, it can't be inline anyway - but generally I avoid inline definitions for anything beyond trivial. 

Right now I'm looking at this purely from how-the-code-looks perspective.

I agree we shouldn't do inline definitions **when** there's a separate .hh and .cc file containing parts of the class.

But in this code, when the whole class sits in a single file, these out-of-line definitions is just repeating ourselves and making the code longer and harder to maintain. The only reason we must do this in general is the .hh vs .cc separation. Rust, for example, doesn't have out-of-line member function definitions at all. We'll also not need them once we migrate away from header files to C++ modules.

Anyway, it's not that important, I don't like the code repetition and the code length, but since it's a purely subjective thing, it's fine if it stays that way.